sbin/celeborn-daemon.sh

```diff
@@ -165,74 +165,87 @@ run_command() {
 
 }
 
-case $option in
-
-  (start)
+start_celeborn() {
     run_command class "$@"
-    ;;
-
-  (stop)
-
-    if [ -f $pid ]; then
-      TARGET_ID="$(cat "$pid")"
-      if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]] || [[ $(ps -p "$TARGET_ID" -o comm=) =~ "jboot" ]]; then
-        echo "stopping $command"
-        kill "$TARGET_ID" && rm -f "$pid"
-      else
-        echo "no $command to stop"
-      fi
-    else
-      echo "no $command to stop"
-    fi
-    ;;
-
-  (restart)
+}
 
+stop_celeborn() {
     if [ -f $pid ]; then
       TARGET_ID="$(cat "$pid")"
       if [[ $(ps -p "$TARGET_ID" -o comm=) =~ "java" ]] || [[ $(ps -p "$TARGET_ID" -o comm=) =~ "jboot" ]]; then
         echo "stopping $command"
-        kill "$TARGET_ID" && rm -f "$pid"
+        kill "$TARGET_ID"
         wait_time=0
         # keep same with `celeborn.worker.graceful.shutdown.timeout`
         wait_timeout=600
         while [[ $(ps -p "$TARGET_ID" -o comm=) != "" && $wait_time -lt $wait_timeout ]];
         do
-          sleep 1s
+          sleep 1
           ((wait_time++))
           echo "waiting for server shutdown, wait for ${wait_time}s"
         done
+
         if [[ $(ps -p "$TARGET_ID" -o comm=) == "" ]]; then
-          run_command class "$@"
+          rm -f "$pid"
         else
-          echo "stopping $command failed."
+          echo "Failed to stop server after ${wait_timeout}s, try 'kill -9 ${TARGET_ID}' forcefully"
+          kill -9 "${TARGET_ID}"
```

Comment:
is it safe when graceful shutdown is enabled?

Reply:
Remove `kill -9`.
Modify `wait_timeout=5` as test.
```bash
./sbin/stop-master.sh
stopping org.apache.celeborn.service.deploy.master.Master
waiting for server shutdown, wait for 1s
waiting for server shutdown, wait for 2s
waiting for server shutdown, wait for 3s
waiting for server shutdown, wait for 4s
waiting for server shutdown, wait for 5s
Failed to stop server(pid=73631) after 5s
```